### PR TITLE
feat(components): add Hours component

### PR DESCRIPTION
**Pull request by tatimblin**
This component will render an hours table from Yext hours. It is accompanied by Hours and HoursInterval classes, which provide useful methods for manipulating and comparing a Yext hours type.

cc @Ceseura

### CODE REVIEW

File: packages/pages/src/components/hours/hours.test.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import { Hours, HoursInterval } from "./hours";
+import { HOURS_WITH_HOLIDAY, HOURS_WITH_REOPEN_DATE } from "./sampleData";
+
+describe("Hours", () => {
+  const hours = new Hours(HOURS_WITH_HOLIDAY);
+  const dateOpen = new Date(2022, 6, 11, 13, 14);
+  const dateClosed = new Date(2022, 6, 11, 22, 10);
+  const dateEndOfWeek = new Date(2022, 6, 17, 13, 14);
+
+  it("return the interval containing a given Date", () => {
```
Comment (mkilpatrick) [inline packages/pages/src/components/hours/hours.test.ts:10]: Same present tense comment for the test names.

File: packages/pages/src/components/hours/hoursTable.tsx [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+ * @param {HoursTableDayData[]} hoursDays
+ * @returns {HoursTableDayData[]} where adjacent days with the same intervals are combined
+ */
+function collapseDays(hoursDays: HoursTableDayData[]): HoursTableDayData[] {
+  let collapsedDays: HoursTableDayData[] = [];
+  hoursDays.forEach((hoursDay) => {
+    const latestGroup = collapsedDays[collapsedDays.length - 1];
+
+    // latestGroup = undefined indicates that this is the first group of days
+    //  add a new 'collapsedDay'
```
Comment (mkilpatrick) [inline packages/pages/src/components/hours/hoursTable.tsx:55]: Extra space before "add"

File: packages/pages/src/components/hours/hoursTable.tsx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    nameMap.wednesday || defaultDayOfWeekNames[3],
+    nameMap.thursday || defaultDayOfWeekNames[4],
+    nameMap.friday || defaultDayOfWeekNames[5],
+    nameMap.saturday || defaultDayOfWeekNames[6],
+  ];
+}
+
+/*
+ * The HoursTable component uses Hours data to generate a table
+ *  listing the business hours of the entity.
```
Comment (mkilpatrick) [inline packages/pages/src/components/hours/hoursTable.tsx:124]: Extra space

File: packages/pages/src/components/hours/hoursTable.tsx [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+ * @param {Intl.DateTimeFormatOptions} timeOptions
+ * @param {String[]} dayOfWeekNames label for each day of week, ordered starting from Sunday
+ * @param {String} startOfWeek set the day of the first row of the table
+ * @param {Boolean} collapseDays combine adjacent rows (days) with the same intervals
+ * @param {Function} intervalStringsBuilderFn override rendering for the interval on each table row
+ */
+const HoursTable: React.FC<HoursTableProps> = (props) => {
+  // Use two rendering passes to avoid SSR issues where server & client rendered content is different
+  //  On the first pass, don't render any content in this component, only set `state.isClient`
+  //  On the second pass (After the page has been loaded), render the content
```
Comment (mkilpatrick) [inline packages/pages/src/components/hours/hoursTable.tsx:136]: Lowercase After

File: packages/pages/src/components/hours/hoursTable.tsx [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+ *
+ * @param {HoursType} hours data from Yext Streams
+ * @param {Intl.DateTimeFormatOptions} timeOptions
+ * @param {String[]} dayOfWeekNames label for each day of week, ordered starting from Sunday
+ * @param {String} startOfWeek set the day of the first row of the table
+ * @param {Boolean} collapseDays combine adjacent rows (days) with the same intervals
+ * @param {Function} intervalStringsBuilderFn override rendering for the interval on each table row
+ */
+const HoursTable: React.FC<HoursTableProps> = (props) => {
+  // Use two rendering passes to avoid SSR issues where server & client rendered content is different
```
Comment (mkilpatrick) [inline packages/pages/src/components/hours/hoursTable.tsx:134]: Add periods at the end of sentences

File: packages/pages/src/components/hours/hoursTable.tsx [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+              </div>
+            );
+          })}
+        </div>
+      )}
+    </>
+  );
+};
+
+export { HoursTable };
```
Comment (mkilpatrick) [inline packages/pages/src/components/hours/hoursTable.tsx:215]: Is this the name you all want to go with vs just calling it Hours? The PMs might have an opinion.

Comment (tatimblin) [reply packages/pages/src/components/hours/hoursTable.tsx:215]: Commented in slackhttps://yextops.slack.com/archives/C02TTUT4C5Q/p1660580587284159

**Review by mkilpatrick — DISMISSED** [review]

**Review by mkilpatrick — DISMISSED** [review]

**Review by bhainesva — APPROVED** [review]

**Pull request merged**